src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingHome.java

```diff
@@ -512,12 +538,47 @@ public void setBuildingLevel(final int level)
         getColony().getCitizenManager().calculateMaxCitizens();
     }
 
+    /**
+     * Gets a list of all beds in this building.
+     * 
+     * @return a list of all beds in this building.
+     */
     @NotNull
     public List<BlockPos> getBedList()
     {
         return new ArrayList<>(bedList);
     }
 
+    /**
+     * Checks whether the given bed is occupied.
+     * 
+     * @param bed the head of the bed to check.
+     * @return whether the given bed is occupied.
+     */
+    public boolean isBedOccupied(BlockPos bed)
+    {
+        return occupiedBeds.contains(bed);
+    }
+
+    /**
+     * Sets whether the given bed is occupied.
```

Comment:
There is a newDay call in here. On which we should reset this list.

Reply:
yes, i didn't notice that method :)